File: android/core/ui/src/main/kotlin/com/dualmusic/core/ui/Effects.kt
package com.dualmusic.core.ui

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Halos lumineux (miroir des `--shadow-glow*` du web) : la signature « néon » de Dual Music.
 *
 * Compose n'a pas de flou de type box-shadow ; on approche l'effet avec une ombre colorée
 * (`spotColor`/`ambientColor` = violet) et une élévation élevée. Rendu proche du glow web.
 */

/** Halo violet standard (éléments interactifs). */
fun Modifier.dmGlow(radius: Int = 12): Modifier = this.shadow(
    elevation = radius.dp,
    shape = RoundedCornerShape(12.dp),
    ambientColor = GlowViolet,
    spotColor = GlowViolet,
)

/** Halo violet intense (éléments actifs/live). */
fun Modifier.dmGlowStrong(): Modifier = dmGlow(radius = 20)

// Violet de marque (280 70% 55%) — constant hors composition pour usage dans les modifiers.
private val GlowViolet = Color.hsl(280f, 0.70f, 0.55f)

File: ios/Packages/CoreUI/Sources/CoreUI/Effects.swift
import SwiftUI

/// Halos lumineux (miroir des `--shadow-glow*` du web) : la signature « néon » de Dual Music.
public extension View {
    /// Halo violet standard. Web : `0 0 40px hsl(280 70% 55% / 0.40)`.
    func dmGlow() -> some View {
        shadow(color: Color(webHSL: 280, 0.70, 0.55, alpha: 0.40), radius: 20)
    }

    /// Halo violet intense (éléments actifs/live). Web : `0 0 60px hsl(280 70% 55% / 0.60)`.
    func dmGlowStrong() -> some View {
        shadow(color: Color(webHSL: 280, 0.70, 0.55, alpha: 0.60), radius: 30)
    }

    /// Ombre portée élégante des cartes. Web : `0 10px 40px -10px hsl(280 70% 55% / 0.30)`.
    func dmElegantShadow() -> some View {
        shadow(color: Color(webHSL: 280, 0.70, 0.55, alpha: 0.30), radius: 20, x: 0, y: 8)
    }
}
